src/handler/mod.rs

```diff
@@ -1192,4 +1364,173 @@ impl Handler {
             .ban_nodes
             .retain(|_, time| time.is_none() || Some(Instant::now()) < *time);
     }
+
+    async fn on_hole_punch_tgt_enr(
+        &mut self,
+        tgt_enr: Option<Enr>,
+        relay_msg_notif: Notification,
+    ) -> Result<(), HolePunchError> {
+        let Some(tgt_enr) = tgt_enr else {
+            return Err(HolePunchError::Relay(Discv5Error::Custom("Target enr not found")));
+        };
+        let tgt_node_address =
+            match NodeContact::try_from_enr(tgt_enr, self.nat_hole_puncher.ip_mode) {
+                Ok(contact) => contact.node_address(),
+                Err(e) => return Err(HolePunchError::Relay(e.into())),
+            };
+        if let Some(session) = self.sessions.cache.get_mut(&tgt_node_address) {
+            trace!(
+                "Sending notif to target {}. relay msg: {}",
```

Comment:
But here notif should probably just be notification, and on second thought ``session.encrypt_notification::<P>(self.node_id, &relay_msg_notif.encode()) {`` just using the full word would be more consistent then having ``notification and ``notif`` on the same line